cedar-policy-core/src/ast/restricted_expr.rs

```diff
@@ -315,7 +315,7 @@ impl<'a> Hash for RestrictedExprShapeOnly<'a> {
 }
 
 /// Errors generated in the restricted_expr module
-#[derive(Debug, Clone, PartialEq, Hash, Error)]
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Hash, Error)]
```

Comment:
Why this change? Are other errors Ser/De?

Reply:
`cedar-policy-core`'s `Response` type was originally Ser/De, so when I nested the error type inside it I needed to make that type Ser/De too (which involved adding `Serialize` and `Deserialize` in a bunch of places). Looking at the code again, I'm not sure that `Response` actually needs to be Ser/De. Possibly for `cedar-spec`? I'll play around with it and get back to you.

Reply:
Getting rid of Ser/De does indeed break DRT, but it looks pretty simple to fix. I haven't looked at FFI yet -- but it's probably a bad to break our serialized representations since they may have wider impact (e.g., on external contributors writing language bindings). Thanks for pointing that out.

It looks like I can use `#[serde_as(as = "DisplayFromStr")]` [[link](https://docs.rs/serde_with/latest/serde_with/struct.DisplayFromStr.html)] to use `Display` for serialization. It expects a `FromStr` for deserialization though. I don't think we want to write that for error messages. Thoughts?

Reply:
Can we get away with having serialization but not deserialization?

Alternatively can we have deserialization use the structure but serialization do strings?  it might be weird that they don't roundtrip, but that might be an arrangement that makes sense in this case for backwards compatibility?  or maybe we don't want to go there

Reply:
Actually, nvm. It looks like our json interface isn't trying to serialize `cedar-policy-core`'s `Response` type. It just exposes the `String` error messages exposed through `cedar-policy`. I think the best approach is to remove Ser/De for `Response`s and make the necessary fixes in DRT.